src/lib/tests/gridsterCompact.service.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { 
+    TestBed,
+    async,
+    fakeAsync,
+    tick,
+    ComponentFixture,
+    ComponentFixtureAutoDetect,
+    inject } from '@angular/core/testing'
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { By } from '@angular/platform-browser';
+
+import { GridsterComponentInterface } from '../gridster.interface';
+import { GridsterComponent } from '../gridster.component';
+import { GridsterCompact } from '../gridsterCompact.service';
+
+describe('gridsterCompact service', () => {
+    let compactType = 'none';
+    let collision: boolean;
+
+    class MockGridsterComponent {
+
+        $options: any = this.getCompactType();
+        grid: any = [{$item: {compactEnabled: false}}, {$item: {compactEnabled: true}}];
+
+        getCompactType() {
+            switch (compactType) {
+                case 'compactUp':
+                    return {compactType: 'compactUp'};
+                case 'compactLeft':
+                    return {compactType: 'compactLeft'};
+                case 'compactUp&Left':
+                    return {compactType: 'compactUp&Left'};
+                case 'compactLeft&Up':
+                    return {compactType: 'compactLeft&Up'};
+                default:
+                    return {compactType: 'none'};
+            }
+        }
+
+        checkCollision(param: any): any {
```

Comment:
here you need a check if item got to exceeded row 0 or column 0
so if $item.row < 0 || $item.column < 0